ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/VolDB.hs

```diff
@@ -194,7 +195,10 @@ getMaxSlotNo db = withSTM db VolDB.getMaxSlotNo
 
 putBlock :: (MonadCatch m, HasHeader blk) => VolDB m blk -> blk -> m ()
 putBlock db@VolDB{..} b = withDB db $ \vol ->
-    VolDB.putBlock vol (extractInfo b) (CBOR.toBuilder (encBlock b))
+    VolDB.putBlock vol (extractInfo (void binaryEnc) b)
+      (CBOR.toBuilder $ binaryBlob binaryEnc)
+  where
+    binaryEnc = encBlock b
```

Comment:
I suggest the following:
```haskell
    VolDB.putBlock vol (extractInfo binInfo b) binaryBlob
  where
    binInfo@BinaryInfo { binaryBlob } = CBOR.toBuilder <$> encBlock b
```
`extractInfo` ignores the binary blob anyway, so no need to `void`.